### PR TITLE
Add Feature Flag for Address Validation Alerts, Convert Unit Tests, and Update Button Component

### PR DESCRIPTION
**Pull request by asiisii**
# PR Title: Add Feature Flag for Address Validation Alerts, Convert Unit Tests, and Update Button Component

## Summary

This PR implements the `profileShowNoValidationKeyAddressAlert` feature flag, which controls the visibility of two different alert messages on the **Contact Information** page's **Address** section, depending on the validation case. Additionally, I converted the unit tests for the `addressValidationView.unit.spec.jsx` component from the deprecated Enzyme library to React Testing Library and added new tests for the updated alert messages and button displays.

Furthermore, in the `addressValidationMessages.js` file, I replaced the standard `button` element with the `va-button` component for consistency with our design system.

These components are used in the **Profile** page but are part of the platform, so they are not directly maintained by the Authenticated Experience Profile team.

### Changes

- **Feature Flag:**
  - Created the `profileShowNoValidationKeyAddressAlert` feature flag to toggle two different alert messages:
    1. **Case 1 (No validationKey and no suggestion):**
       - **Headline:** "This address you entered is invalid"
       - **Message:** "We can’t confirm the address you entered with the U.S. Postal Service. You’ll need to go back to edit it."
       - Only the primary button "Go back to edit" is shown.
       - [Figma Design](https://www.figma.com/design/bFdl7MEIda4ExZIQuot84r/Profile---Contact-Information?node-id=926-6559&t=DdF5a8Nd8s5oCnbg-4)
    2. **Case 2 (No validationKey but has suggestion):**
       - **Headline:** "We can’t confirm the address you entered with the U.S. Postal Service"
       - **Message:** "We can use the suggested address we found. Or, you can go back to edit the address you entered."
       - Two buttons are shown:
         - Primary button: "Go back to edit"
         - Secondary button: "Use suggested address"
       - [Figma Design](https://www.figma.com/design/bFdl7MEIda4ExZIQuot84r/Profile---Contact-Information?node-id=926-7252&t=I2btxQkThRXFdf5V-4)

- **Unit Tests:**
  - Converted unit tests for the `addressValidationView.unit.spec.jsx` component from Enzyme to React Testing Library.
  - Added new tests to validate the correct alert messages and button displays based on the validation case.

- **Button Component:**
  - Replaced the standard `button` element with the `va-button` component in `addressValidationMessages.js` for consistency with our design system.

## Related Issue(s)

- **Ticket:** department-of-veterans-affairs/va.gov-team#93272

## Testing Done

- **Unit Tests:**
  - Converted from Enzyme to React Testing Library.
  - Added new tests for different alert scenarios and ensured the correct buttons are displayed.

- **Manual Testing:**
  - Verified both cases for the alert messages and button configurations are working correctly based on the feature flag and validation state.

## What areas of the site does it impact?

- Profile > Contact Information > Address section

## Acceptance Criteria

1. The correct alert message and buttons should be displayed based on whether the address has no `validationKey` and no suggestion or has a suggestion.
2. The `va-button` component should replace the standard `button` element across the impacted files.
3. The new tests should pass, confirming that the feature flag is working as expected and that the buttons are displayed correctly.

## Quality Assurance & Testing

- [x] I updated and converted unit tests for the feature flag and alert messages.
- [x] Replaced the `button` element with `va-button` for consistency.
- [x] No sensitive information (i.e., PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs.
- [x] Linting warnings have been addressed.
- [ ] Documentation has been updated ([link to documentation](#) *if necessary).
- [ ] Screenshot of the developed feature is added.
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed.

### Error Handling

- [x] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution.
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable).

### Authentication

- [x] Did you log in to a local build and verify all authenticated routes work as expected with a test user?

### CODE REVIEW

File: src/applications/personalization/profile/mocks/endpoints/address.js [hunk 1]
```diff
@@ -107,7 +107,7 @@ const addressValidation = {
         zipCode: '94115',
       },
       addressMetaData: {
-        confidenceScore: 0,
+        confidenceScore: 88,
```
Comment (nickjg231) [inline src/applications/personalization/profile/mocks/endpoints/address.js:110]: I see this is a mock, but what's the purpose for changing this?

File: src/platform/user/profile/vap-svc/constants/addressValidationMessages.js [hunk 2]
```diff
@@ -7,6 +7,8 @@ export const ADDRESS_VALIDATION_TYPES = Object.freeze({
   MISSING_UNIT_OVERRIDE: 'missingUnitNumberOverride',
   SHOW_SUGGESTIONS_OVERRIDE: 'showSuggestionsOverride',
   SHOW_SUGGESTIONS_NO_CONFIRMED_OVERRIDE: 'showSuggestionsNoConfirmedOverride',
+  NO_SUGGESTIONS_NO_OVERRIDE: 'noSuggestionsNoOverride',
+  SHOW_SUGGESTIONS_NO_OVERRIDE: 'showSuggestionsNoOverride',
```
Comment (nickjg231) [inline src/platform/user/profile/vap-svc/constants/addressValidationMessages.js:11]: What are the different use cases for `SHOW_SUGGESTIONS_NO_CONFIRMED_OVERRIDE` and `SHOW_SUGGESTIONS_NO_OVERRIDE`, and could these be combined?

Comment (nickjg231) [reply src/platform/user/profile/vap-svc/constants/addressValidationMessages.js:11]: I see in the next file, never mind!

**Review by nickjg231 — APPROVED** [review]
Just a question but otherwise looks good!

File: src/applications/personalization/profile/mocks/endpoints/address.js (hunk 1 above)
Comment (asiisii) [reply src/applications/personalization/profile/mocks/endpoints/address.js:110]: I was testing with different confidenceScore to see if UI changes somehow to better understand how the address validation worked @nickjg231

**Review by rmessina1010 — APPROVED** [review]
LGTM

**Pull request merged**